tsdb/head.go

```diff
@@ -524,28 +532,59 @@ Outer:
 					seriesCreationErr = err
 					break Outer
 				}
+				if !created {
```

Comment:
For all these changes in the `loadWAL`, please review https://github.com/prometheus/prometheus/pull/7438 (same changes in different PR as it's not totally related to snapshotting)